hummingbot/connector/gateway_EVM_AMM.py

```diff
@@ -613,8 +614,79 @@ async def _update_balances(self, on_interval=False):
             self._in_flight_orders_snapshot = {k: copy.copy(v) for k, v in self._in_flight_orders.items()}
             self._in_flight_orders_snapshot_timestamp = self.current_timestamp
 
+    async def execute_cancel(self, trading_pair: str, client_order_id: str) -> str:
+        self._general_cancel(client_order_id, None)
+
+    async def _general_cancel(self, order_id: str, cancel_age: Optional[int] = None) -> str:
+        try:
+            tracked_order = self._in_flight_orders.get(order_id)
+            if tracked_order is None:
+                self.logger().error(f"The order {order_id} is not tracked. ")
+                raise ValueError
+
+            # if cancel_age is included, then cancel only if the age of the order
+            # is geq to the cancel_age
+            if cancel_age is not None:
+
+                if (self.current_timestamp - tracked_order.creation_timestamp).total_seconds() < cancel_age:
+                    return
+
+            if tracked_order.is_done or tracked_order.is_cancelling:
+                return
+
+            tracked_order.last_state == "CANCELING"
+
+            self.logger().info(f"Requesting cancel of order {order_id}")
+
+            await gateway_http_client.cancel_evm_transaction(
+                self.chain,
+                self.network,
+                self.address,
+                tracked_order.nonce)
+
+            tracked_order.last_state == "CANCELED"
+
+            self.logger().info(f"Order {order_id} has been canceled")
+
+            return order_id
+
+        except Exception as err:
+            self.logger().error(
+                f"Failed to cancel order {order_id}: {str(err)}.",
+                exc_info=True
+            )
+
+    async def cancel_outdated_orders(self, cancel_age: int):
+        self.cancel_all(30.0, cancel_age)
+
     async def cancel_all(self, timeout_seconds: float) -> List[CancellationResult]:
-        return []
+        self._general_cancel_all(timeout_seconds, None)
```

Comment:
This would cause all active orders to be cancelled when the user stops the bot.
I think it's best to leave as it was due to the peculiarity of gateway transaction.
That way users don't have to waste fees just for stopping the bot, and will be good for cases where users have several orders yet to be filled before stopping the bot

Reply:
Ideally, you should let the orders continue on the blockchain but keep the in fight orders recorded in SQLite via the tracking states system. So the next time Hummingbot starts, the connector can check for the statuses of those, and see if there's a need to cancel them.

Now... the tracking states system doesn't seem to be well followed and looks disused to me. So that's another problem.

Reply:
Also, missing await, and the return value.